dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/PreCheckExistenceValidationHookTest.java

```diff
@@ -505,6 +508,22 @@ void verifyRelationshipValidationFailsWhenIsDeleteAndRelationshipIsNotPresent()
         hasTrackerError( reporter, E4016, RELATIONSHIP, rel.getUid() );
     }
 
+    @Test
+    void verifyRelationshipValidationFailsWhenIsSoftDeleted()
+    {
+        // given
+        Relationship rel = Relationship.builder()
+            .relationship( SOFT_DELETED_RELATIONSHIP_UID )
```

Comment:
I prefer keeping the test fixtures local if possible. So I would do create a relationship using `softDeletedRelationship(someUid)` here, pass that into the `rel` and the mock.

Reply:
Done